staging/src/k8s.io/component-helpers/apimachinery/lease/controller.go

```diff
@@ -80,6 +81,28 @@ func NewController(clock clock.Clock, client clientset.Interface, holderIdentity
 		client:                     client,
 		leaseClient:                leaseClient,
 		holderIdentity:             holderIdentity,
+		leaseName:                  holderIdentity,
+		leaseNamespace:             leaseNamespace,
+		leaseDurationSeconds:       leaseDurationSeconds,
+		renewInterval:              renewInterval,
+		clock:                      clock,
+		onRepeatedHeartbeatFailure: onRepeatedHeartbeatFailure,
+		newLeasePostProcessFunc:    newLeasePostProcessFunc,
+	}
+}
+
+// NewControllerWithLeaseName is a copy of NewController but accepts a leaseName parameter.
+// Use this constructor in cases when the lease name and holder identity should be different.
+func NewControllerWithLeaseName(clock clock.Clock, client clientset.Interface, holderIdentity string, leaseDurationSeconds int32, onRepeatedHeartbeatFailure func(), renewInterval time.Duration, leaseName, leaseNamespace string, newLeasePostProcessFunc ProcessLeaseFunc) Controller {
```

Comment:
Alternative approach: add a `WithLeaseName()` method to the controller.

Don't feel too strongly and fine with either approach.

Reply:
I like breaking it